### ISSUE TITLE
TODO: add a dev note regarding sudoless docker

### ISSUE THREAD

**Issue by Tranquility2** [testcontainers/testcontainers-python#749]
Devs should be informed (using the official docs) that the testing environment considers `sudoless docker` as a baseline (and some things might not work / break if sudo is required) 

For a future plan, we should rewrite / avoid using docker command directly (for example using `subprocess.run`) and use the docker SDK.

**Comment by Tranquility2** [comment]
![Image](https://github.com/user-attachments/assets/11c3655d-1fd6-4fc2-8b47-f672878990d1)
This is the general gist

**Comment by Tranquility2** [comment]
@alexanderankin @totallyzen please tell me how you think is best to proceed and I'll open a PR accordingly.

**Comment by alexanderankin** [comment]
what did you do that didnt work? for example, i just pulled latest main, ran tests with `make tests`, expected that to work and it did. Well, I'm still waiting - currently it is running aws, core/arango succeeded.

**Comment by Tranquility2** [comment]
Checkout the comments on https://github.com/testcontainers/testcontainers-python/pull/714/files#r1885389659
TLDR: the dind tests fails as they run "docker build" without sudo using `subprocess`.

**Comment by Tranquility2** [comment]
I assume you prefer not to add this comment so I'll start working on replacing the relevant code.

**Comment by Tranquility2** [comment]
Created https://github.com/testcontainers/testcontainers-python/pull/750/files

**Comment by alexanderankin** [comment]
ah okay so "sudoless" refers not to rootless docker or anything but rather it refers to the permissions on the docker socket.

I consider it invalid docker installation if you do not also do

```
sudo usermod -aG docker $USER
```

after installing.

I wonder what the difference is between using `DockerClient()` - as I've instructed people to do the usermod and that has unblocked people using testcontainers in general before on linux (via slack).

**Comment by Tranquility2** [comment]
Note that its not the user that affect the behavior but the complimentary requirement 
`export DOCKER_HOST=unix:///run/user/1000/docker.sock` [source](https://docs.docker.com/engine/security/rootless/) 
This can be avoided as the `DockerClient` fetch the `DOCKER_HOST` and make our life easier without the need to relay or case if docker is running with/without sudo :)
